### PR TITLE
DOC: update wavefile.py to fix `write` example in docs

### PR DESCRIPTION
**Pull request by trevorandersen**
#### Reference issue
n/a

#### What does this implement/fix?
This fixes the example code provided in the docstring for the write function. Firstly, this imports numpy so the code can be run as-is. Secondly, the data is converted to np.int16 so that the WAV file is created correctly as 100 Hz sine wave.

### CODE REVIEW

File: scipy/io/wavfile.py [hunk 1]
```diff
@@ -742,10 +742,11 @@ def write(filename, rate, data):
     Write to 16-bit PCM, Mono.
 
     >>> from scipy.io.wavfile import write
+    >>> import numpy as np
```
Comment (ilayn) [inline scipy/io/wavfile.py:745]: This is always assumed on SciPy docs to avoid writing hundreds of identical lines, hence needs to be omitted

Comment (trevorandersen) [reply scipy/io/wavfile.py:745]: Thanks ilayn, sorry for my ignorance.

Comment (ilayn) [reply scipy/io/wavfile.py:745]: No problem at all. Thank you for your time.

File: scipy/io/wavfile.py [hunk 2]
```diff
@@ -742,10 +742,11 @@ def write(filename, rate, data):
...
     >>> samplerate = 44100; fs = 100
     >>> t = np.linspace(0., 1., samplerate)
     >>> amplitude = np.iinfo(np.int16).max
-    >>> data = amplitude * np.sin(2. * np.pi * fs * t)
+    >>> data = np.int16(amplitude * np.sin(2. * np.pi * fs * t))
     >>> write("example.wav", samplerate, data.astype(np.int16))
```
Comment (peterbell10) [inline scipy/io/wavfile.py:750]: Isn't `astype` here doing the same thing already?

Comment (trevorandersen) [reply scipy/io/wavfile.py:750]: It certainly is. I'm not sure how I missed that initially. Sorry for my confusion.

**Pull request closed without merging**